src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,432 @@
+'use client';
+
+import { Availability } from '@/types/Availability';
+import React, { useState, useEffect, useCallback } from 'react';
+
+type ClientScheduleAppointmentProps = {
+    availabilities: Availability[];
+};
+
+export default function ClientScheduleAppointment({ availabilities }: ClientScheduleAppointmentProps) {
+    const MONTH_NAMES = [
+        'Janeiro',
+        'Fevereiro',
+        'Março',
+        'Abril',
+        'Maio',
+        'Junho',
+        'Julho',
+        'Agosto',
+        'Setembro',
+        'Outubro',
+        'Novembro',
+        'Dezembro',
+    ];
+
+    const DAYS = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];
+
+    const [month, setMonth] = useState(new Date().getMonth());
+    const [day, setDay] = useState(new Date().getDate());
+    const [year, setYear] = useState(new Date().getFullYear());
+    const [no_of_days, setNumDays] = useState<number[]>([]);
+    const [blankdays, setBlankDays] = useState<number[]>([]);
+    const [datepickerValue, setDatepickerValue] = useState('');
+    const [selectedHour, setSelectedHour] = useState<number | null>();
+
+    const [step, setStep] = useState(0);
+    const [availabilitiesMap, setAvailabilities] = useState<Map<string, number[]>>();
+    const [availableHours, setAvailableHours] = useState<number[]>();
+
+    // Calculates the number of days in the given month
+    const getNoOfDays = useCallback(
+        (month: number) => {
+            let daysInMonth = new Date(year, month + 1, 0).getDate();
+            let dayOfWeek = new Date(year, month).getDay();
+
+            let blankdaysArray: number[] = [];
+            for (var i = 1; i <= dayOfWeek; i++) {
+                blankdaysArray.push(i);
+            }
+
+            let daysArray: number[] = [];
+            for (var i = 1; i <= daysInMonth; i++) {
+                daysArray.push(i);
+            }
+
+            setBlankDays(blankdaysArray);
+            setNumDays(daysArray);
+        },
+        [year],
+    );
+
+    useEffect(() => {
+        getNoOfDays(month);
+    }, [month, getNoOfDays]);
```

Comment:
I get the purpose of the useEffect here, but its a calculation that could be done beforehand and not have this useEffect re-rendering the page... 

Can you try to get rid of it?

My suggestion is to use something like https://date-fns.org/v2.30.0/docs/isBefore to map the dates of the month https://date-fns.org/v2.30.0/docs/getDaysInMonth to get the blank and num days as you are doing on `getNoOfDays`